lib/segment/src/index/field_index/map_index/mod.rs

```diff
@@ -620,36 +646,140 @@ impl PayloadFieldIndex for MapIndex<UuidIntType> {
         &'a self,
         condition: &'a FieldCondition,
     ) -> Option<Box<dyn Iterator<Item = PointOffsetType> + 'a>> {
-        if let Some(Match::Value(MatchValue {
-            value: ValueVariants::Keyword(keyword),
-        })) = &condition.r#match
-        {
-            let keyword = keyword.as_str();
-
-            if let Ok(uuid) = Uuid::from_str(keyword) {
-                return Some(Box::new(self.get_iterator(&uuid.as_u128()).copied()));
-            }
-        }
+        match &condition.r#match {
+            Some(Match::Value(MatchValue { value })) => match value {
+                ValueVariants::String(uuid_string) => {
+                    let uuid = Uuid::from_str(uuid_string).ok()?;
+                    Some(Box::new(self.get_iterator(&uuid.as_u128()).copied()))
+                }
+                ValueVariants::Integer(_) => None,
+                ValueVariants::Bool(_) => None,
+            },
+            Some(Match::Any(MatchAny { any: any_variant })) => match any_variant {
+                AnyVariants::Strings(uuids_string) => {
+                    let uuids: Result<IndexSet<u128>, _> = uuids_string
+                        .iter()
+                        .map(|uuid_string| Uuid::from_str(uuid_string).map(|x| x.as_u128()))
+                        .collect();
 
-        None
+                    let uuids = uuids.ok()?;
+
+                    Some(Box::new(
+                        uuids
+                            .into_iter()
+                            .flat_map(|uuid| self.get_iterator(&uuid).copied())
+                            .unique(),
+                    ))
+                }
+                AnyVariants::Integers(integers) => {
+                    if integers.is_empty() {
+                        Some(Box::new(iter::empty()))
+                    } else {
+                        None
+                    }
```

Comment:
this and similar cases reasoning: we can't guaratnee which type will be used for de-serialization of empty list, as it is ambiguous. Having check for empty condition is a fail-safe